### ISSUE TITLE
BottomTabs Animations API

### ISSUE THREAD

**Issue by alan-nakagiri** [wix/react-native-navigation#6340]
### Issue Description

I've been looking to find how to animate the bottomTabs. When I apply an alpha animation to the content on push and pop and hide the bottom tabs, the bottom tabs still use the native translationX animation on iOS and translationY animation on Android. From what I've read it looks like I need apply a bottomTab animation s `animations -> hide/show -> bottomTabs`, but that did not seem to work. 

I've attempted multiple variations of how to apply an animation to bottomTabs and have not figured it out. I cannot tell if this is a bug or I am missing something. 

Can someone please help clarify where/how I need to apply an animation for the bottomTabs to also fade when visibility toggled during pop/push?

### Steps to Reproduce / Code Snippets / Screenshots

Here's a snippet of my options (bottomTabs animation not defined because I know my solution is wrong)

```
options: {
 bottomTabs: { visible: false }, 
 animations: {
    push: {
      content: {
        alpha: {
          from: 0,
          to: 1,
          duration: 275,
        },
      },
    },
    pop: {
      content: {
        alpha: {
          from: 0,
          to: 1,
          duration: 275,
        },
      },
    },
 }
}
```

**iOS**
![Screen Recording 2020-06-26 at 10 14 23 AM mov](https://user-images.githubusercontent.com/11300359/85883798-6c68b400-b796-11ea-8018-71dcab2d54e7.gif)

**Android**
![Screen Recording 2020-06-26 at 10 51 07 AM mov](https://user-images.githubusercontent.com/11300359/85886473-09c5e700-b79b-11ea-8c0f-477da2f6a551.gif)

---
### Environment
* React Native Navigation version: 6.7.5
* React Native version: 0.61.1
* Platform(s) (iOS, Android, or both?): Both
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator, Debug

**Comment by guyca** [comment]
@alan-nakagiri The animation API in its current state is still lacking functionality to support the use case you presented. That being said, I think you're better of using showModal instead of push. This will allow you to easily fade in the new screen, without the need to mess around with BottomTabs animation.

**Comment by alan-nakagiri** [comment]
@guyca Got it. Thanks for the update! It looks like you guys have made a number of strides recently. Also, I appreciate the work you've been doing maintaining this library. We've been using it in production for a few years now and it's worked out pretty well for us 👍

**Comment by mrousavy** [comment]
Same issue here. Using a Modal seems like a weird hack for this to work (😕) but at least it works 😄

I've tried the following, but haven't found a way to animate the bottom bar at all...

```ts
return Navigation.push(componentId, {
	component: {
		options: {
			animations: {
				push: {
					bottomTabs: {
						alpha: {
							from: 1,
							to: 0,
							duration: 150,
							interpolation: 'accelerate',
						},
						translationY: {
							from: 0,
							to: 150,
							duration: 150,
						},
						translationX: undefined,
						x: undefined,
					},

```

**Comment by nenti** [comment]
@alan-nakagiri How did you get the big center button on the bottom tab on iOS to work?

**Comment by alan-nakagiri** [comment]
@nenti The icon is sized at 60x60 with 1x, 2x 3x sizes included in our assets and it has `iconInsets` defined as `{ top: -10, left: 0, bottom: 0, right: 0 }`. It's nested pretty deep into the stack.

So `bottomTabs -> children -> stack -> children -> component -> options -> bottomTab -> iconInsets`

**Comment by nenti** [comment]
Thank you so much! I love that and sorry for highjacking your bug thread. Last question: What do you mean 1x 2x 3x? From my understanding only hard images are allowed as icons right?

**Comment by mrousavy** [comment]
Hi! I've come up with a potential solution:

![ezgif com-video-to-gif-3](https://user-images.githubusercontent.com/15199031/93451068-fd06ea00-f8d6-11ea-8b05-aac6068f357c.gif)

That's how you animate a tabBar: 

Before the push command ([here](https://github.com/wix/react-native-navigation/blob/c8c925abb0c0162095e4671a15133396260e56d7/lib/ios/UINavigationController%2BRNNCommands.m#L17)) add the following:
```objc
    CGRect frame = self.tabBarController.tabBar.frame;
    CGFloat height = frame.size.height;
    CGFloat offsetY = [newTop hidesBottomBarWhenPushed] ? -height : height;

    CGFloat duration = (animated)? 0.3 : 0.0;

    [UIView animateWithDuration:duration animations:^{
        self.tabBarController.tabBar.frame = CGRectOffset(frame, 0, offsetY);
    }];
```

This will run the animation when the new viewcontroller's `hidesBottomBarWhenPushed` resolves to true. I've disabled that for every view controller to avoid the default animation, not sure if that's required or not. Anyway, this is obviously only a placeholder, so the real animation API parsing (from the JS `StackAnimationOptions.bottomTabs` object) should happen here and run the animation in parallel with the screen pushing. Same for popping.

One problem with this: As you can see, my TabBar has a button that overflows the top, that's why the code sample above doesn't really work for me, since then the tip of the purple button is still visible. I've multiplied the `height` times `1.1` as a dirty hack, so a solution for that would also be required... (Especially when we implement #6522)

**Comment by mrousavy** [comment]
So apparently animating the tabBar is a really hard thing to do on iOS and should be avoided. I have no idea how other native apps do it, but I have a proposal: Make the bottomTabs not animate at all when pushing to a screen that has `bottomTabs: { visible: false }` AND a custom `push` animation, but rather render the pushed screen above the bottom tabs. That's how it works in `react-navigation`, you nest a Tab Navigator inside a Stack, which leads to the same result.

**Comment by Brad19** [comment]
I'm looking forward to do something that is already done with react-navigation. Is there anyway to add it something like below
https://www.npmjs.com/package/react-native-animating-bottom-tab-bar

Any updates would be really appreciated

**Comment by mrousavy** [comment]
@Brad19 The difference between react-navigation and react-native-navigation is that react-native-navigation uses the _native_ bottom tabs APIs. This means, that it looks, acts and feels different on iOS and android, since the bottom tabs are not react components. While this provides a lot of benefits for look and feel, stability and performance, this also means that there is limited flexibility when it comes to stuff like animating the bar, especially animations like the one you linked. So no, it is not possible at the moment, and I don't think it will ever be. 

If you absolutely need this implemented, you can implement the tab navigator logic yourself using React. It's easier than implementing screen transitions 🤷. Or you can try to implement these animations using the native platforms, so once for iOS and once for Android. But that's far more complex.

**Comment by Brad19** [comment]
@mrousavy Thanks for the update. I get the context. I'll have to decide if I can do this in React. But it would be nice to have if there is an option.

**Comment by yedidyak** [comment]
Closing this as part of a maintainer cleanup of open issues older than one year.

Decision: Closed: stale/non-critical; no evidence it is a critical current-version RNN issue.

If this still reproduces on the latest React Native Navigation with a minimal reproduction, please open a new issue with current RN/RNN versions and repro steps.
